compiler/modules/front.syntax/syntaxNode.bal

```diff
@@ -539,6 +539,12 @@ function syntaxNodeFromFunctionTypeDesc(FunctionTypeDesc td, boolean functionSig
                                      retTd != () ? [{ token: "returns" }, syntaxNodeFromTypeDesc(retTd)] : ());
 }
 
+function syntaxNodeFromRestParam(FunctionTypeParam restParam) returns SubSyntaxNode {
+    return nonTerminalSyntaxNode(restParam, syntaxNodeFromTypeDesc(restParam.td),
+                                            { token: "..." },
+                                            { name: <string>restParam.name, pos: <Position>restParam.namePos });
```

Comment:
This doesn't look right to me. Why are you guaranteed a `name` field here.

I suggest you combine this with syntaxNodeFromFunctionTypeParam.

Reply:
fixed